notoptions-mitigation.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Plugin Name: VIP NotOptions Mitigation
+ * Description: Detect invalid values in the notoptions cache.
+ * Author: Automattic
+ * License: GPL version 2 or later - http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+ */
+
+namespace Automattic\VIP;
+
+use Automattic\VIP\Utils\Alerts;
+
+if ( is_admin() || ( defined( 'WP_CLI' ) && WP_CLI ) ) {
+	add_action( 'muplugins_loaded', __NAMESPACE__ . '\notoptions_mitigation', -999 );
+}
+
+function notoptions_mitigation() {
+	$notoptions = wp_cache_get( 'notoptions', 'options' );
+
+	// filter for any values not equal to (bool)true.
+	$not_trues = array_filter( $notoptions, function( $v ) {
+		return $v !== true;
+	} );
+
+	// if they exist, something's borked
+	if ( $not_trues ) {
```

Comment:
Can we more explicitly check this?
```suggestion
	if ( 1 >= count( $not_trues ) ) {
```

Reply:
Actually we count it further down with `$total_invalid`, so we can just move that up here and check against it.